packages/angular/src/wallet/wallet.store.ts

```diff
@@ -0,0 +1,295 @@
+import { Inject, Injectable, Optional } from '@angular/core';
+import { ComponentStore } from '@ngrx/component-store';
+import { SendTransactionOptions, WalletNotConnectedError, WalletNotReadyError } from '@solana/wallet-adapter-base';
+import { WalletName } from '@solana/wallet-adapter-wallets';
+import { Connection, Transaction } from '@solana/web3.js';
+import { asyncScheduler, combineLatest, defer, from, Observable, of, throwError } from 'rxjs';
+import { catchError, concatMap, filter, first, map, observeOn, switchMap, tap, withLatestFrom } from 'rxjs/operators';
+
+import { fromAdapterEvent, isNotNull } from '../operators';
+import { WalletNotSelectedError } from './wallet.errors';
+import { WALLET_CONFIG } from './wallet.tokens';
+import { WalletConfig, WalletState } from './wallet.types';
+
+export const WALLET_DEFAULT_CONFIG: WalletConfig = {
+    wallets: [],
+    autoConnect: false,
+    localStorageKey: 'walletName',
+    onError: (error: unknown) => console.error(error),
+};
+
+@Injectable()
+export class WalletStore extends ComponentStore<WalletState> {
+    private readonly _autoConnect = this._config?.autoConnect || false;
```

Comment:
Should this be readonly? I can't necessarily think of a case where the dapp would want to change this value, but I'm hesitant to enforce that at the type level.

Reply:
TBH I wouldn't expect consumers to change the config on runtime but maybe there's a use-case I'm not seeing. I just add readonly unless I'm sure I'll be changing it. At the time of writing I assumed consumers wouldn't be changing config afterward.

It's an easy change though.

Reply:
Yeah, hard for me to say. It's private anyway so it's nonbreaking to change this later. However it's worth noting that the react package exposes the value of `autoConnect` in the context so that descendants can detect whether it's enabled or not.